qa/suites/orch/cephadm/workunits/task/test_set_mon_crush_locations.yaml

```diff
@@ -0,0 +1,61 @@
+roles:
+- - host.a
+  - osd.0
+  - mon.a
+  - mgr.a
+- - host.b
+  - osd.1
+  - mon.b
+  - mgr.b
+- - host.c
+  - osd.2
+  - mon.c
+tasks:
+- install:
+- cephadm:
+- cephadm.apply:
+    specs:
+      - service_type: mon
+        service_id: foo
+        placement:
+          count: 3
+        spec:
+          crush_locations:
+            host.a:
+              - datacenter=a
+            host.b:
+              - datacenter=b
+              - rack=2
+            host.c:
+              - datacenter=a
+              - rack=3
+- cephadm.shell:
+    host.a:
+      - |
+        set -ex
+        # since we don't know the real hostnames before the test, the next
+        # bit is in order to replace the fake hostnames "host.a/b/c" with
+        # the actual names cephadm knows the host by within the mon spec
+        ceph orch host ls --format json | jq -r '.[] | .hostname' > realnames
+        echo $'host.a\nhost.b\nhost.c' > fakenames
+        echo $'{datacenter=a}\n{datacenter=b,rack=2}\n{datacenter=a,rack=3}' > crush_locs
+        ceph orch ls --service-name mon --export > mon.yaml
+        MONSPEC=`cat mon.yaml`
+        echo "$MONSPEC"
+        while read realname <&3 && read fakename <&4; do
+          MONSPEC="${MONSPEC//$fakename/$realname}"
+        done 3<realnames 4<fakenames
+        echo "$MONSPEC" > mon.yaml
+        cat mon.yaml
+        # now the spec should have the real hostnames, so let's re-apply
+        ceph orch apply -i mon.yaml
+        sleep 90
+        ceph orch ps --refresh
+        ceph orch ls --service-name mon --export > mon.yaml; ceph orch apply -i mon.yaml
+        sleep 90
+        ceph mon dump
+        ceph mon dump --format json
+        # verify all the crush locations got set from "ceph mon dump" output
+        while read hostname <&3 && read crushloc <&4; do
+          ceph mon dump --format json | jq --arg hostname "$hostname" --arg crushloc "$crushloc" -e '.mons | .[] | select(.name == $hostname) | .crush_location == $crushloc'
```

Comment:
```
2023-04-04T02:44:03.615 INFO:teuthology.orchestra.run.smithi012.stderr:+ ceph mon dump --format json
2023-04-04T02:44:03.615 INFO:teuthology.orchestra.run.smithi012.stderr:+ jq --arg hostname smithi012 --arg crushloc '{datacenter=a}' -e '.mons | .[] | select(.name == $hostname) | .crush_location == $crushloc'
2023-04-04T02:44:03.620 INFO:journalctl@ceph.mon.c.smithi172.stdout:Apr 04 02:44:03 smithi172 ceph-mon[92425]: from='client.? 172.21.15.12:0/3654628154' entity='client.admin' cmd=[{"prefix": "mon dump"}]: dispatch
2023-04-04T02:44:04.004 INFO:teuthology.orchestra.run.smithi012.stderr:dumped monmap epoch 6
2023-04-04T02:44:04.300 INFO:journalctl@ceph.mon.a.smithi012.stdout:Apr 04 02:44:04 smithi012 ceph-mon[121586]: pgmap v188: 1 pgs: 1 active+clean; 577 KiB data, 20 MiB used, 268 GiB / 268 GiB avail
2023-04-04T02:44:04.300 INFO:journalctl@ceph.mon.a.smithi012.stdout:Apr 04 02:44:04 smithi012 ceph-mon[121586]: from='client.? 172.21.15.12:0/160266058' entity='client.admin' cmd=[{"prefix": "mon dump", "format": "json"}]: dispatch
2023-04-04T02:44:04.300 INFO:journalctl@ceph.mon.a.smithi012.stdout:Apr 04 02:44:04 smithi012 ceph-mon[121586]: from='client.? 172.21.15.12:0/3953290221' entity='client.admin' cmd=[{"prefix": "mon dump", "format": "json"}]: dispatch
2023-04-04T02:44:04.373 DEBUG:teuthology.orchestra.run:got remote process result: 4
2023-04-04T02:44:04.375 ERROR:teuthology.run_tasks:Saw exception from tasks.
```
forgot the mons are stored by their id, not hostname. Typically for cephadm deployments these are the same, but in this case the IDs are actually "a", "b", and "c"